src/tests/system/tests/test_gpo.py

```diff
@@ -0,0 +1,822 @@
+"""
+SSSD GPO tests
+
+:requirement: IDM-SSSD-REQ: Group policy host based access control
+
+The following features and parameters are not covered:
+- ad_gpo_cache_timeout
+- ad_gpo_map_network
+- ad_gpo_map_batch
+- ad_gpo_map_service
+- ad_gpo_map_permit
+- ad_gpo_map_deny
+"""
+
+from __future__ import annotations
+
+import pytest
+from sssd_test_framework.roles.ad import AD
+from sssd_test_framework.roles.client import Client
+from sssd_test_framework.topology import KnownTopology
+
+
+@pytest.mark.importance("critical")
+@pytest.mark.topology(KnownTopology.AD)
+def test_gpo__is_set_to_enforcing(client: Client, ad: AD):
+    """
+    :title: Group policy object host base access control is set to enforcing
+    :description:
+        When ad_gpo_access_control is set to enforcing, user access is managed and enforced by the policy.
+        Unlisted users are denied access.
+    :setup:
+        1. Create the following users; 'user', 'user1', 'user2', 'deny_user1', 'deny_user2'
+        2. Create the following groups; 'group' with member 'user2' and 'deny_group' with member 'deny_user2'
+        3. Create and link the GPO 'site policy' and add 'user1', group and 'Domain Admins' to
+           SeInteractiveLogonRight key. Add 'deny_user1' and 'deny_group' to SeDenyInteractiveLogonRight key
+        4. Configure sssd.conf with 'ad_gpo_access_control' = 'enforcing'
+        5. Start SSSD
+    :steps:
+        1. Authenticate 'user1' and 'user2' with both su and ssh
+        2. Authenticate 'user', 'deny_user' and 'deny_user1' with both su and ssh
+    :expectedresults:
+        1. User authentication is successful
+        2. User authentication is unsuccessful
+    :customerscenario: True
+    """
+    ad.user("user").add()
+    user1 = ad.user("user1").add()
+    user2 = ad.user("user2").add()
+    deny_user1 = ad.user("deny_user1").add()
+    deny_user2 = ad.user("deny_user2").add()
+    group = ad.group("group").add().add_members([user2])
+    deny_group = ad.group("deny_group").add().add_members([deny_user2])
+
+    ad.gpo("site policy").add().policy(
+        {
+            "SeInteractiveLogonRight": [user1, group, ad.group("Domain Admins")],
+            "SeDenyInteractiveLogonRight": [deny_user1, deny_group],
+        }
+    ).link()
+
+    client.sssd.domain["ad_gpo_access_control"] = "enforcing"
+    client.sssd.start()
+
+    assert client.auth.ssh.password(username="user1", password="Secret123")
+    assert client.auth.su.password(username="user1", password="Secret123")
+    assert client.auth.ssh.password(username="user2", password="Secret123")
+    assert client.auth.su.password(username="user2", password="Secret123")
+    assert not client.auth.ssh.password(username="user", password="Secret123")
```

Comment:
This one got a lot of assertions maybe split it it to two tests with allowed user and denied user.

Reply:
Okay, I'll break up the test cases.